src/SqlProviderBase.ts

```diff
@@ -306,23 +308,35 @@ export abstract class SqlProviderBase extends NoSqlProvider.DbProvider {
                             fieldList.push('nsp_pk TEXT PRIMARY KEY');
                             fieldList.push('nsp_data TEXT');
 
-                            const columnBasedIndexes = _.filter(storeSchema.indexes, index =>
+                            const columnBasedIndices = _.filter(storeSchema.indexes, index =>
                                 !indexUsesSeparateTable(index, this._supportsFTS3));
-                                
-                            const indexColumns = _.map(columnBasedIndexes, index => 'nsp_i_' + index.name + ' TEXT');
+
+                            const indexColumns = _.map(columnBasedIndices, index => 'nsp_i_' + index.name + ' TEXT');
                             fieldList = fieldList.concat(indexColumns);
                             const tableMakerSql = 'CREATE TABLE ' + storeSchema.name + ' (' + fieldList.join(', ') + ')';
                             
                             const currentIndexMetas = _.filter(indexMetadata, meta => meta.storeName === storeSchema.name);
+                            
+                            // find indices in the schema that did not exist before
+                            const newIndices = _.filter(storeSchema.indexes, index => {
+                                const indexIdentifier = getIndexIdentifier(storeSchema, index);
+                                return !_.some(currentIndexMetas, meta => meta.key === indexIdentifier);
+                            });
+
+                            // find indices in the meta that do not exist in the new schema
+                            const removedIndexMetas = _.filter(currentIndexMetas, meta => 
+                                !_.some(storeSchema.indexes, newIndex => getIndexIdentifier(storeSchema, newIndex) === meta.key));
```

Comment:
Suggestion: You are calling `getIndexIdentifier` on each `storeSchema.indexes` twice. Also, each of these two `filter` are `O(n*m)`. You could get this down to `O(n+m)` by getting each `indexIdentifier` and `meta.key` once and put them into dictionaries.